### PR TITLE
TeamCity Improvements

### PR DESCRIPTION
**Pull request by BlythMeister**
### Description

* Correcting TeamCity publish named artifacts
* Add report build problems
* Tidy TeamCity listener
* Extract all messages into TeamCityInternal (which is internal...as the name suggests) and called by TeamCity (which is public).  Added advantage is that messages can be used in many ways by different public methods.  Also, the messages suddenly become strongly typed(ish)

### CODE REVIEW

File: src/app/Fake.BuildServer.TeamCity/TeamCity.fs [hunk 1]
```diff
@@ -412,9 +377,6 @@ module TeamCity =
                     match description with
                     | Some d -> TeamCityWriter.sendOpenBlock tag.Name (sprintf "%s: %s" tag.Type d)
                     | _ -> TeamCityWriter.sendOpenBlock tag.Name tag.Type
-                | TraceData.CloseTag (tag, _, TagStatus.Failed) ->
```
Comment (matthid) [inline src/app/Fake.BuildServer.TeamCity/TeamCity.fs]: So we don't need CloseTag?

Comment (BlythMeister) [reply src/app/Fake.BuildServer.TeamCity/TeamCity.fs]: This case for CloseTag with a status of Failed is exactly the same as the CloseTag below with all TagStatus.

**Comment by matthid** [comment]
Hm, I tried to merge in latest release/next changes into this PR and now apparently github cannot handle the diff anymore :(

**Comment by BlythMeister** [comment]
I'll take a look in the morning 🙂

**Comment by BlythMeister** [comment]
i have rebased this (and all my other PRs)

**Comment by matthid** [comment]
Thanks I thought stuff would go green again but this azure pipeline thing doesn't seem to be well thought through...

**Comment by BlythMeister** [comment]
:( that's a shame...

**Comment by matthid** [comment]
Thanks!

**Pull request merged**